Address concern about confusing words and ownership of resource code (#19729)
dotnet/docs: docs/standard/garbage-collection/implementing-dispose.md

# Implement a Dispose method

> Implementing the <xref:System.IDisposable.Dispose%2A> method is primarily for releasing unmanaged resources. When working with instance members that are <xref:System.IDisposable> implementations, it's common to cascade <xref:System.IDisposable.Dispose%2A> calls. There are additional reasons for implementing <xref:System.IDisposable.Dispose%2A>, such as undoing something that was previously done. For example, freeing memory that was allocated, removing an item from a collection that was added, signaling the release of a lock that was acquired, and so on.

Reviewer: ```suggestion
Implementing the <xref:System.IDisposable.Dispose%2A> method is primarily for releasing unmanaged resources. When working with instance members that are <xref:System.IDisposable> implementations, it's common to cascade <xref:System.IDisposable.Dispose%2A> calls. There are additional reasons for implementing <xref:System.IDisposable.Dispose%2A>, for example, freeing memory that was allocated, removing an item that was added to a collection, or signaling the release of a lock that was acquired.
```

One minor suggestion in the last sentence.
